src/solver/simulation/sim_structure_probleme_economique.h

```diff
@@ -707,4 +715,41 @@ struct PROBLEME_HEBDO
     double maxPminThermiqueByDay[366];
 };
 
+// hourly CSR problem structure
+class HOURLY_CSR_PROBLEM
+{
+private:
+    void calculateCsrParameters();
+    void resetProblem();
+    void buildProblemVariables();
+    void setVariableBounds();
+    void buildProblemConstraintsLHS();
+    void buildProblemConstraintsRHS();
+    void setProblemCost();
+    void solveProblem(uint week, int year);
+public:
+    void run(uint week, int year);
+    
+    int hourInWeekTriggeredCsr;
+    double belowThisThresholdSetToZero;
+    PROBLEME_HEBDO* pWeeklyProblemBelongedTo;
+    HOURLY_CSR_PROBLEM(int hourInWeek, PROBLEME_HEBDO* pProblemeHebdo)
+    {
+        hourInWeekTriggeredCsr = hourInWeek;
+        pWeeklyProblemBelongedTo = pProblemeHebdo;
+        belowThisThresholdSetToZero
+          = pProblemeHebdo->adqPatchParams->ThresholdCSRVarBoundsRelaxation;
+    };
+    std::map<int, int> numberOfConstraintCsrEns;
+    std::map<int, int> numberOfConstraintCsrAreaBalance;
+    std::map<int, int> numberOfConstraintCsrFlowDissociation;
+    std::map<int, int> numberOfConstraintCsrHourlyBinding; // length is number of binding constraint
```

Comment:
The 3 previous map are all used inside the CSR quadratic problem.
They should be data member of an object of class **csrQuadraticProblem** (see remarks in **adequacy_patch.cpp**).
Same remark for **varToBeSetToZeroIfBelowThreshold**